### DISCUSSION TITLE
Wrong Username or Password

### DISCUSSION THREAD

**Discussion by DavTechFr** [samemory/homebridge-eufy-security-control#279] [Q&A] [answered]
Hello Dear community,

I am very sorry to annoy you with this issue.

But when I provide the Eufy credentials in the plugin splash screen , result is “Wrong username or password” :(

I’ve tried with Primary account, or family one (Admin rights).
email/password are correct as I can log to iPhone App using them.

The log is (credentials have been removed)
…

24/11/2021, 22:20:53] [Homebridge UI] [homebridge-eufy-security] Incoming Request: /request-otp
[24/11/2021, 22:20:53] [Homebridge UI] [homebridge-eufy-security] [1.0.4] DEBUG: Load previous login_hash: 054e818a19016130b404dc79b5fxxxxx
[24/11/2021, 22:20:53] [Homebridge UI] [homebridge-eufy-security] [1.0.4] DEBUG: Generated new openudid: de4ec635ebdd1e48
[24/11/2021, 22:20:53] [Homebridge UI] [homebridge-eufy-security] [1.0.4] DEBUG: Generated new serial_number: d489998e7fcc
[24/11/2021, 22:20:53] [Homebridge UI] [homebridge-eufy-security] [1.0.4] DEBUG: Authenticate and get an access token { token: null, tokenExpiration: null }
[24/11/2021, 22:20:53] [Homebridge UI] [homebridge-eufy-security] [1.0.4] DEBUG: Request: {
  method: 'post',
  endpoint: 'passport/login',
  baseUrl: 'https://mysecurity.eufylife.com/api/v1',
  token: null,
  data: {
    email: '03inxxxxx@icloud.com',
    password: 'xxxxxx-xxxxxx-xxxxxx'
  },
  headers: {
    app_version: 'v2.8.0_887',
    os_type: 'android',
    os_version: '30',
    phone_model: 'EUFYCLIENT',
    country: 'US',
    language: 'en',
    openudid: 'de4ec635ebdd1e48',
    uid: '',
    net_type: 'wifi',
    mnc: '02',
    mcc: '262',
    sn: 'd489998e7fcc',
    Model_type: 'PHONE',
    timezone: 'GMT+01:00'
  }
}
[24/11/2021, 22:20:54] [Homebridge UI] [homebridge-eufy-security] [1.0.4] DEBUG: Response: 
[24/11/2021, 22:20:54] [Homebridge UI] [homebridge-eufy-security] [1.0.4] ERROR: Status return code not 200 { status: 204, statusText: 'No Content' }
[24/11/2021, 22:20:54] [Homebridge UI] [homebridge-eufy-security] [1.0.4] ERROR: Token error
[24/11/2021, 22:21:06] [Homebridge UI] [homebridge-eufy-security] Terminating child process...
[24/11/2021, 22:21:06] [Homebridge UI] [homebridge-eufy-security] Child process ended
…

What is my mistake?

Thanks,
David

**Comment by MattC3156** [comment]
Same issue here with same debug log.  Have tried TFP on and off, reinstall of plugin, resinstall of Homebridge (have moved from iOS to Pi so starting new installation).  Account checked, tried guest and normal but still same issue.

**Comment by DavTechFr** [comment]
Hi,

I’ve tried this morning same Family credentials with another plugin (without 2FA) :
https://github.com/birkir/homebridge-plugin-eufy-security

This Plugin succeeds to connect.

So I think there **may have issue with the content (headers?) of the request sent to Eufy portal** :/

**How can I help?**

David

**Comment by schliemann** [comment]
This is a bug in the client library

**Comment by schliemann** [comment]
It has been fixed. But we need to use the newest version.

**Reply by DavTechFr** [reply]
Hello,

Ah OK, good news.
When this version will be available?

is there a workaround in meantime?
(Downgrade to previous versions,…)

David

**Comment by DavTechFr** [comment, marked as answer]
New version 1.0.6 fixes the issue.
thanks a lot
David

**Comment by Gadagadanagadanagana** [comment]
I have the same issue now on 1.0.8....

**Comment by schliemann** [comment]
@Laeae that no devices are found?

**Reply by Gadagadanagadanagana** [reply]
sorry, that my username and password are not able to get me logged in. Error message "Wrong Username or Password", despite correct information. 

In my case it seems that this may be due to some former login configuration, that is still saved on my raspberry, even when I have removed and reinstalled the plugin. 

[23/12/2021, 09:33:48] [Homebridge UI] [homebridge-eufy-security] Incoming Request: /request-otp
[23/12/2021, 09:33:48] [Homebridge UI] [homebridge-eufy-security] [1.0.8] DEBUG: Load previous login_hash: b37e621998311ed22da43e9338aa1dfe
[23/12/2021, 09:33:48] [Homebridge UI] [homebridge-eufy-security] [1.0.8] INFO:  Authentication properties changed, invalidate saved cloud token.

**Comment by schliemann** [comment]
Have you created a guest user without 2FA?

**Reply by Gadagadanagadanagana** [reply]
I have switched off 2FA for my existing user, but not created a guest user. 

My config file is definately correct.

**Reply by schliemann** [reply]
It is a bad idea to use your primary user since eufy might block it. Try and create a user solely for the plugin.

**Reply by Gadagadanagadanagana** [reply]
Maybe this is the case. Is there some best practice then to request access again, or transfer my cameras to a different account? 

Thanks for any hints in this situation

**Reply by schliemann** [reply]
Go to Family and guests settings

![B997855E-618A-434E-A9C3-B15850AB58D6](https://user-images.githubusercontent.com/10818044/147220547-74d97cb8-6423-43a3-b6e1-b13fe0bd0227.jpeg)

**Reply by Gadagadanagadanagana** [reply]
Thank you SO MUCH, after a lot of back and forth, it has now worked! Thanks for the "Christmas gift" so to speak!

All the best

**Reply by schliemann** [reply]
You're welcome. Merry Christmas to you 😉

**Comment by schliemann** [comment]
@Laeae And is your config file correct?

**Reply by Haglerd** [reply]
Wait... So are we going to the latest(1.08) or the older 1.06? I was hoping the the 1.08 would allow this to actually work and stream, but still no luck. So I guess if I need to go back to 1.0.6 for this to fix the username. I might as well just keep waiting. 1.0.6 doesn't stream for me.

**Reply by Haglerd** [reply]
Just for a quick follow up. I have now tried 1.1.1-beta.2, 1.0.8, 1.0.6, 1.0.6-alpha.0.2-alpha, none of these let me login. I get the same username password incorrect issue. I've reinstalled, and restarted the server multiple times. No luck. I even changed the eufy password and verified I could login on my phone. So I have the right password. Anybody else got around this?

**Reply by schliemann** [reply]
Have you disabled 2FA?

**Reply by schliemann** [reply]
99% sure something is wrong with your user.

**Reply by Haglerd** [reply]
How can that be the case when I can login just fine on my phone and/or "https://mysecurity.eufylife.com/#/login"? If my user account was messed up wouldn't I not be able to login anywhere else as well? 

I never had 2FA on to begin with and it did work prior to me trying to upgrade to 1.0.8 today. 

Neither version has reliably streamed for me though.

**Reply by schliemann** [reply]
Have you created a guest account for the plugin? Otherwise do that.

**Reply by Haglerd** [reply]
Yes, that is the account I'm using.

**Reply by schliemann** [reply]
For the app as well?

**Reply by Haglerd** [reply]
Yes, that's what I just mentioned previously. The account works everywhere but in the plugin and it only recently stopped working in the plugin. Once I tried to update it. which resulted in me uninstalling and reinstalling... then getting that error.

**Reply by schliemann** [reply]
Please share your logs then.

**Comment by schliemann** [comment]
1.0.8 is the newest non beta and has somewhat functional streaming from cloud. Beta has working local streaming - however it reboots hb on streaming.

**Reply by Haglerd** [reply]
Unfortunately I can't get around the "bad username" issue or I'd give it a shot. Did y'all ever figure that one out? 

Edit: Looks like the Newest version available fixes it. Back in Business!
